scripts/jenkins/cloud/ansible/roles/list_packages/tasks/packages_from_old_jobs.yml

```diff
@@ -28,66 +28,90 @@
   ignore_errors: true
 
-- name: Download package lists from old jobs
-  get_url:
-    url: "https://ci.suse.de/job/openstack-{{ cloud_product }}/{{ item.0.key }}/artifact/{{ item.1.relativePath }}"
-    dest: "{{ diff_tmp_dir }}/rpms/old_jobs/{{ item.0.key }}/{{ item.1.relativePath | basename }}"
-  delegate_to: localhost
-  when:
-    - (item.1.relativePath | dirname).endswith('rpms')
-    - (item.1.relativePath | basename).startswith('rpms_')
-  loop: "{{ _old_jobs_values | dict2items | subelements('value.artifacts')}}"
+- name: Set Facts - old jobs no exists
+  set_fact:
+    _old_jobs_no_exist: false
+  loop: "{{ _output.results }}"
   loop_control:
-    loop_var: item
-  ignore_errors: true
-  register: _downloaded_files
+      label: "{{ item._ansible_item_label }}"
+  when: item.rc == 0
 
-- set_fact:
-    _tmpRelPathVal: "{{ _tmpRelPathVal + [ item.item.0.key|string + ':' + item.item.1.relativePath|basename ] }}"
-  loop:   "{{ _downloaded_files.results }}"
-  when: item is not skipped
-  loop_control:
-      label: "{{ item.item.1.relativePath }}"
 
-- set_fact:
-    _currentJobFileNames: "{{ _currentJobFileNames + [item] }}"
-  loop: "{{ [cloud_env]+groups['cloud_virt_hosts']|flatten(levels=1) }}"
+# skip when no old jobs defined or no old jobs found
+- block:
 
-- name:
-  combine_files:
-    filenames1: "{{_tmpRelPathVal }}"
-    filenames2: "{{ _currentJobFileNames }}"
-  register: _all_items
+    - name: Download package lists from old jobs
+      get_url:
+        url: "https://ci.suse.de/job/{{ job_base_name }}/{{ item.0.key }}/artifact/{{ item.1.relativePath }}"
+        #url: "https://ci.suse.de/job/openstack-{{ cloud_product }}/{{ item.0.key }}/artifact/{{ item.1.relativePath }}"
+        dest: "{{ diff_tmp_dir }}/rpms/old_jobs/{{ item.0.key }}/{{ item.1.relativePath | basename }}"
+      delegate_to: localhost
+      when:
+        - (item.1.relativePath | dirname).endswith('rpms')
+        - (item.1.relativePath | basename).startswith('rpms_')
+      loop: "{{ _old_jobs_values  | dict2items | subelements('value.artifacts')}}"
+      loop_control:
+        loop_var: item
+      ignore_errors: true
+      register: _downloaded_files
 
-- name: Create a diff of installed packages and save into a file in yaml
-  diff_yaml:
-    file1: "{{ diff_tmp_dir }}/rpms/old_jobs/{{ item_virtual_hosts.jobNumber }}/{{ item_virtual_hosts.currentFilename}}"
-    file2: "{{ diff_tmp_dir }}/rpms/{{ item_virtual_hosts.currentFilename }}"
-    output: "{{ diff_tmp_dir }}/rpms/diffs/{{ item_virtual_hosts.jobNumber }}_{{item_virtual_hosts.currentFilename}}"
-  delegate_to: localhost
-  register: diff_files
-  loop: "{{ _all_items.combine_files }}"
-  loop_control:
-    loop_var: item_virtual_hosts
-  when: _all_items.combine_files is not none
+    - name: Set Facts - _temporary Relative Path Value from jobs
+      set_fact:
+        _tmpRelPathVal: "{{ _tmpRelPathVal + [ item.item.0.key|string + ':' + item.item.1.relativePath|basename ] }}"
```

Comment:
_tmpRelPathVal is not really consistent with the usual ansible fact naming scheme of all lower case with `_` delimitation of words. I'm aware that it is technically inherited from previous version of code just more of an observation

Reply:
thx for heads up, I ll change that